src/expandProperty.js

```diff
@@ -130,3 +130,28 @@ export default function expandProperty(property, value) {
     return parseBorder(value.toString(), borderExpand[property])
   }
 }
+
+export default function preExpand(property, value) {
+  if (Array.isArray(value)) {
+    const result = {}
+
+    value.forEach(item => {
+      const itemResult = expandProperty(property, item)
+
+      if (itemResult) {
+        Object.keys(itemResult).forEach(itemProperty => {
+          result[itemProperty] = result[itemProperty] || []
+          result[itemProperty].push(itemResult[itemProperty])
+        })
+      }
+    })
+
+    if (Object.keys(result).length) {
+      return result
+    }
+
+    return null
+  }
+
+  return expandProperty(property, value)
```

Comment:
To be honest I hate this solution 💩 But I don't better ideas. At least I added proper UTs 😼

Reply:
Well for now its fine right? We can improve it later on anyways.